shared/lib/components.mjs

```diff
@@ -28,13 +28,39 @@ export function nunjucksEnv(searchPaths = [], nunjucksOptions = {}) {
 /**
  * Load single component data (from source)
  *
- * @param {string} componentName - Component name
+ * @param {string} component - Component directory name
  * @returns {Promise<ComponentData>} Component data
  */
-export async function getData(componentName) {
-  return import(
-    join(paths.pkg, `src/nhsuk/components/${componentName}/macro-options.mjs`)
+export async function getData(component) {
+  const optionsPath = join(
+    paths.pkg,
+    `src/nhsuk/components/${component}/macro-options.mjs`
   )
+
+  // Bypass import cache (e.g. Gulp watch changes)
+  const options = /** @type {Omit<ComponentData, "component">} */ (
+    await import(`${optionsPath}?imported=${Date.now()}`)
+  )
+
+  // Add component directory name to options
+  return { component, ...options }
+}
+
+/**
+ * Load all component data (from source)
+ */
+export async function getDataList() {
```

Comment:
[very minor] could these two function names be more descriptive? Perhaps `loadComponent` / `loadComponentList` ?